### PR TITLE
feat(jsx/dom): introduce react-dom/client APIs and React.version

### LINKED ISSUE #2792: Missing pieces for React compatibility
### version

In React we can get the version string as follows, but not in hono/jsx

```ts
import { version } from 'react';
```

https://github.com/facebook/react/blob/main/packages/shared/ReactVersion.js#L15

### react/dom/client

The createRoot and hydrateRoot APIs are not implemented in hono.

```ts
import { createRoot, hydrateRoot } from 'react-dom/client';
```

### PR DESCRIPTION
**Pull request by usualoma**
Fixes #2792

### version

There seems to be a library that does conditional branching on the major version number of `React.version`.

This is like a UserAgent of a web browser (e.g. Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537. 36), which is a string for compatibility and somewhat unpleasant to use, and we do not recommend its use, but if this improves compatibility, it should be added.

### createRoot / hydrateRoot

Some options are not yet supported, but they can be rendered in the way recommended by React since version 18, as shown in the following diff.

https://github.com/usualoma/hono-react-compat-demo/commit/38773555179ef54f8564ca1963720ac16f720987

### The author should do the following, if applicable

- [x] Add tests
- [x] Run tests
- [x] `bun run format:fix && bun run lint:fix` to format the code

### CODE REVIEW

**Comment by usualoma** [comment]
@yusukebe How about including this in v4.4?

File: src/jsx/dom/client.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { useState } from '../hooks'
```
Comment (yusukebe) [inline src/jsx/dom/client.ts:1]: Add a [module doc](https://jsr.io/docs/writing-docs#module-documentation) at the top, like #2796.

**Review by yusukebe — COMMENTED** [review]
I've left a comment. And can you update the `exports` field in `jsr.json`?

**Comment by yusukebe** [comment]
> @yusukebe How about including this in v4.4?

Nice! Let's include this.

File: src/jsx/dom/client.ts (hunk 1 above)
Comment (usualoma) [reply src/jsx/dom/client.ts:6]: OK, I added it in 61e2f7e.

**Comment by yusukebe** [comment]
@usualoma 

One thing. Please update the `exports` field in [the `jsr.json`](https://github.com/honojs/hono/blob/next/jsr.json)!

**Comment by usualoma** [comment]
I'm fixing it, just give me a minute.

**Comment by usualoma** [comment]
Sorry I'm late, fixed. 2e0dbca

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@usualoma 

Thank you! Now, I'll merge this into the next. The `v4.4` will be released tomorrow!

**Pull request merged**